runner/s3_import_dynamodb.go

```diff
@@ -61,7 +62,7 @@ func (r S3ImportDynamoDBRunner) Run() (types.CompletionWatcher, error) {
 	r.Logger.Infow("Staring S3 import to DynamoDB materialization runner", "name", r.ID.Name, "variant", r.ID.Variant)
 
 	if r.IsUpdate {
-		return nil, fmt.Errorf("materialization updates are not implemented for S3 import to DynamoDB")
+		return nil, fferr.NewInternalError(fmt.Errorf("materialization updates are not implemented for S3 import to DynamoDB"))
```

Comment:
This is because it should be checked before it reaches here right?

Reply:
Are you saying you'd like me to move this check into the factory? I could do that. It actually makes more sense.